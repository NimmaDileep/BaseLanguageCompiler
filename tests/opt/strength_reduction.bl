var x: Int = 10;
var y: Int = x + 0;
print(y);

var p: Int = 10;
var q: Int = 0 + p;
print(q);

var a: Int = 5;
var b: Int = a - 0;
print(b);

var c: Int = 5;
var d: Int = 0 - c;
print(d);

var r: Int = 15;
var s: Int = r * 0;
print(s);

var t: Int = 15;
var u: Int = 0 * t;
print(u);

var e: Int = 15;
var f: Int = e * 1;
print(f);

var j: Int = 15;
var k: Int = 1 * j;
print(k);

var m: Int = 5;
var n : Int = m * 2;
print(n);

var g: Int = 10;
var h: Int = 2 * n;
print(h);


