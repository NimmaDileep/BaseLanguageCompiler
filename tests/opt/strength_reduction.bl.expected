10
10
5
-5
0
0
15
15
10
20